teuthology/task/ansible.py

```diff
@@ -250,12 +254,28 @@ def _build_args(self):
         return args
 
     def teardown(self):
+        if self.config.get("cleanup"):
+            self._cleanup()
         if self.generated_inventory:
             os.remove(self.inventory)
         if self.generated_playbook:
             os.remove(self.playbook_file.name)
         super(Ansible, self).teardown()
 
+    def _cleanup(self):
+        """
+        If the ``cleanup`` key exists in config the same playbook will be
+        ran again during the teardown step with the var ``teardown`` given with
```

Comment:
Also, this docstring makes it sounds like the function doesn't do anything if 'cleanup' isn't set. I think we should either clarify the docstring or make ``_cleanup()`` itself do the config check - if you don't have strong feelings, I think I prefer the latter.